prep/go/tutorial/chapter4/xkcd.go

```diff
@@ -0,0 +1,465 @@
+package main
+
+import (
+        "os"
+        "fmt"
+        "strconv"
+        "strings"
+        "log"
+        "encoding/json"
+        "net/http"
+        "io/ioutil"
+        "sync"
+        "unicode"
+        "bufio"
+        "sort"
+)
+
+type XKCDComic struct {
+    Month string `json:"month"`
+    Id int `json:"num"`
+    Link string `json:"link"`
+    Year string `json:"year"`
+    News string `json:"new"`
+    SafeTitle string `json:"safe_title"`
+    Transcript string `json:"transcript"`
+    Alt string `json:"alt"`
+    Img string `json:"img"`
+    Title string `json:"title"`
+    Day string `json:"day"`
+    Indexed bool 
+}
+
+type XKCDComicView struct {
+    Id int `json:"id"`
+    Title string `json:"title"`
+    TranscriptSummary string `json:"transcript_summary"`
+    Link string `json:"link"`
+}
+
+type ComicIndex map[string]map[int]bool
+
+func newComicIndex() ComicIndex {
+    var index ComicIndex
+    index = make(map[string]map[int]bool)
+    return index
+}
+
+func addComicIndex(index ComicIndex, term string, id int) {
+    if index[term] == nil {
+        index[term] = make(map[int]bool)
+    }
+    index[term][id] = true
+}
+
+
+func toView(comic XKCDComic) XKCDComicView {
+    var view XKCDComicView
+    view.Id = comic.Id
+    view.Title = comic.Title
+    // Some comics don't have transcripts
+    if len(comic.Transcript) >= 100 {
+        view.TranscriptSummary = comic.Transcript[:100]
+    } else {
+        view.TranscriptSummary = comic.Transcript
+    }
+    view.TranscriptSummary += "..."
```

Comment:
Nit: Did you intend for this only in the truncation case?

Reply:
I always truncate. Something that made it easy to glance about the output but should be user decided. The check is because a few comics don't have transcripts so I can't assume I can transcript[:100] as they might not be 100 characters in length.